realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -155,9 +135,100 @@ public void multipleReferencesFromParentList() {
         assertEquals(parent, child.getListParents().last());
     }
 
+    // A notification callback should be called on a commit that adds a backlink
+    @Test
+    @RunTestInLooperThread
+    public void notification_onCommit() {
+        Realm looperThreadRealm = looperThread.realm;
+
+        looperThreadRealm.beginTransaction();
+        AllJavaTypes child = looperThreadRealm.createObject(AllJavaTypes.class, 10);
+        looperThreadRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+                looperThread.testComplete();
```

Comment:
This should only be called when the test is completed successfully, so something is not right with the asserts and this.